File: ClothingStoreApp/ClothingStore.Application/DTOs/VariantDto.cs
namespace ClothingStore.Application.DTOs
{
    public record class VariantDto
    {
        public Guid VariantId { get; set; }
        public Guid ItemId { get; set; }
        public ColorDto Color { get; set; }
        public int TotalQuantity { get; set; }
        public IEnumerable<AvailableLocationBySizeDto> AvailableLocations { get; init; } = Array.Empty<AvailableLocationBySizeDto>();
        public IEnumerable<PhotoDto> Gallery { get; init; } = Array.Empty<PhotoDto>();
    }
}

File: ClothingStoreFrontend/src/Types/VariantDto.ts
import type { ColorDto } from "./ColorDto";
import type { PhotoDto } from "./PhotoDto";
import type { AvailableLocationBySizeDto } from "./AvailableLocationBySizeDto";

export interface VariantDto{
    variantId: string;
    itemId: string;
    color: ColorDto;
    totalQuantity: number;

    availableLocations: AvailableLocationBySizeDto[];
    gallery: PhotoDto[];
}